src/content/general/architcture/openstack/index.md

```diff
@@ -0,0 +1,164 @@
+
+### Bootstrapping
+
+The initial deployment entails the creation of that management cluster in a defined region. After the management cluster is ready we deploy all our automation taking advantage of Kubernetes primitives and using the same philosophy we advocate to our customers.
+
+### Components
+
+Giant Swarm leverages the concept of “Operators" to control all resources that clusters need as “Custom Resources”. At the same time customers can also use the Kubernetes Control Plane API to manage their clusters and/or applications.
+
+#### Organization operator
+
+This operator is in charge of reconcile `Organization` custom resources. The functionality of the operator is pretty straightforward, just take care of create and delete organization namespace when the given resource is created.
+
+To know more about organization please read [this documentation](https://docs.giantswarm.io/general/organizations/). 
+
+#### RBAC operator
+
+We have created RBAC operator with the goal of maintain permissions up to date between the different organization and users on the management cluster.
+
+#### App operator
+
+The [App Platform](https://docs.giantswarm.io/app-platform/) is a system we have created to deliver app in multi cluster fashion. App operator is the main piece running in management cluster to manage App custom resource and make sure the applications and configuration are delivered correctly across the different workload clusters.
+
+#### Cluster API operators
+
+In [Cluster API]() there is a set of generic operators that all Cluster API providers runs to address the common bootstrap and management actions on a Kubernetes cluster lifecycle. There is `Kubeadm Bootstrap`controller which is in charge of provider a cloud init configuration to turn a machine into a Kubernetes node. Also there is a `Kubeadm Control Plane` controller that manages the lifecycle of the control plane nodes and provide access to the API. And finally we have `CAPI manager` controller that manages cluster and machine resources.
+
+#### Cluster OpenStack controller
+
+A part of the generic controllers we have an special one that act as a bridge and reconcile the provider specific configuration to create the necessary infrastructure for a Kubernetes cluster. This controller provision, update and delete all resources on the provider API.
+
+#### Cluster apps operator
+
+This simple operator takes care of the default apps has to be created for a cluster and the right configuration that need to be created. Assets like the certificate authority or the domain base are provided by it via a configmap.
+
+## Giant Swarm on-premises workload cluster(s)
+
+### Components
+
+#### Cloud Provider OpenStack
+
+This is the controller that is deployed inside all OpenStack clusters in order to provide Kubernetes integration. It listen to services and ingresses to provide load balancers dynamically for the workload clusters. 
```

Comment:
```suggestion
This is the controller that is deployed inside all OpenStack clusters in order to provide Kubernetes integration. It listen to services and ingresses to provide load balancers dynamically for the workload clusters. It also enriches Kubernetes node info with infrastructure relevant information. This is helpful for application administrators to implement affinity/ anti-affinity rules which reflect underlying availability zone infos. It's also mandatory for the Cluster API provisioning workflow to get known infrastructure specific machine infos.
```

Reply:
And i would also recomment to add `CSI` in this list and then describe the openstack integration.